solver/src/interactions/allowances.rs

```diff
@@ -24,11 +24,7 @@ pub trait AllowanceManaging {
 
     /// Returns the approval interaction for the specified token and spender for
     /// at least the specified amount.
-    async fn get_approval(&self, token: H160, spender: H160, amount: U256) -> Result<Approval> {
-        self.get_allowances(hashset![token], spender)
-            .await?
-            .approve_token(token, amount)
-    }
+    async fn get_approval(&self, token: H160, spender: H160, amount: U256) -> Result<Approval>;
```

Comment:
This change was required because `async_trait` generates code with `where Self: ...` bounds that aren't trait object safe.